rust/src/render.rs

```diff
@@ -196,28 +203,28 @@ pub fn render(options: Options) -> RenderedString {
 
 	if !options.spaceless {
 		match options.align {
-			Align::Top => output.push(String::from("\n\n\n")),
-			Align::Bottom => output[0] = String::from("\n\n\n\n") + &output[0],
+			Align::Top => output.push(format!("{line_break}{line_break}{line_break}")),
```

Comment:
I just had a look at `format!("{line_break}{line_break}{line_break}")` vs `line_break.repeat(3).to_string()` in terms of performance but it all sort of washes away because of the `.to_string()` call. I wonder if making `line_break` a `String` from the start (a single heap allocation) and then just use `.repeat` where you need it?
But then we don't have that many occurrences here that actually need this.
Never mind me... probably just getting distracted. _(I thought it was still worth sharing)_

Reply:
I do agree it's pretty ugly, but I don't think there's a big difference here. The repeat option is cleaner, you could also store the amount of repeats as a variable and modify that really easily in the future if you wanted. Would you prefer that?

Reply:
I don't think you need to do anything in this PR. Overall I need to redo how I handle strings in this library. I think it's wildly inefficient. When I wrote this rust port I was locked in a hotel room in Toronto because of covid for 10 days and may have also hallucinated things 😬 
Using the `format` macro should be kept to a monimum and there is probably a better way to deal with how we build those strings by using a single mutable reference. One day I have time for this :)

Reply:
Fixing the tests is the only outstanding thing for this to land

Reply:
Are there tests that are not passing? On my local machine, `cargo test` is passing all tests.

Reply:
I'm talking about the CI tests. GitHub actions are failing because some of them might be outdated. I'll try to fix it and then we can merge that fix into this branch and merge to main.